diff --git a/configure.ac b/configure.ac
index 689a3f7..95e05e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -99,10 +99,7 @@ AC_CHECK_HEADERS([ssl.h])
 AC_CHECK_LIB(dl, dlopen)
 AC_CHECK_LIB(rt, clock_gettime)
 AC_CHECK_LIB(crypt, crypt)
-AC_CHECK_LIB([lua], lua_callk, , [have_liblua=no], [-ldl -lm])
-if test x$have_liblua = xno; then
-   AC_CHECK_LIB([lua5.1], lua_call, , AC_MSG_ERROR(Required liblua missing), [-ldl -lm])
-fi
+AC_SEARCH_LIBS(lua_call, [lua lua5.1], , AC_MSG_ERROR(Required liblua missing), [-ldl -lm])
 AC_CHECK_LIB(talloc, talloc_named_const, , AC_MSG_ERROR(Required libtalloc missing) )
 AC_CHECK_HEADERS([talloc.h talloc/talloc.h])
 
